webapp/graphite/tags/redis.py

```diff
@@ -151,29 +156,44 @@ def get_series(self, path, requestContext=None):
 
     return TaggedSeries(tags['name'], tags)
 
-  def list_tags(self, tagFilter=None, requestContext=None):
-    return sorted([
+  def list_tags(self, filter=None, limit=None, requestContext=None):
+    results = sorted([
       {'tag': tag}
       for tag in self.r.sscan_iter('tags')
-      if not tagFilter or re.match(tagFilter, tag) is not None
+      if not filter or re.match(filter, tag) is not None
     ], key=lambda x: x['tag'])
 
-  def get_tag(self, tag, valueFilter=None, requestContext=None):
+    if limit:
+      return results[:limit]
+
+    return results
+
+  def get_tag(self, tag, filter=None, limit=None, requestContext=None):
     if not self.r.sismember('tags', tag):
       return None
 
     return {
       'tag': tag,
-      'values': self.list_values(tag, valueFilter=valueFilter),
+      'values': self.list_values(
+        tag,
+        filter=filter,
+        limit=limit,
+        requestContext=requestContext
+      ),
     }
 
-  def list_values(self, tag, valueFilter=None, requestContext=None):
-    return sorted([
+  def list_values(self, tag, filter=None, limit=None, requestContext=None):
+    results = sorted([
       {'value': value, 'count': self.r.scard('tags:' + tag + ':values:' + value)}
       for value in self.r.sscan_iter('tags:' + tag + ':values')
-      if not valueFilter or re.match(valueFilter, value) is not None
+      if not filter or re.match(filter, value) is not None
     ], key=lambda x: x['value'])
 
+    if limit:
```

Comment:
kind of similar like above, why not apply the limit earlier and save a lot of dicts?

Reply:
good call, will update